pubnub-gson/src/main/java/com/pubnub/api/PubNub.java

```diff
@@ -0,0 +1,228 @@
+
+    @NotNull DeleteMessages deleteMessages();
+
+    @NotNull MessageCounts messageCounts();
+
+    @NotNull Grant grant();
+
+    @NotNull GrantTokenBuilder grantToken();
+
+    @NotNull
+    GrantTokenBuilder grantToken(Integer ttl);
+
+    @NotNull RevokeToken revokeToken();
+
+    @NotNull GetState getPresenceState();
+
+    @NotNull SetState setPresenceState();
+
+    @NotNull Publish publish();
+
+    @NotNull Signal signal();
+
+    @NotNull ListAllChannelGroup listAllChannelGroups();
+
+    @NotNull AllChannelsChannelGroup listChannelsForChannelGroup();
+
+    @NotNull AddChannelChannelGroup addChannelsToChannelGroup();
+
+    @NotNull RemoveChannelChannelGroup removeChannelsFromChannelGroup();
+
+    @NotNull DeleteChannelGroup deleteChannelGroup();
+
+    SetUUIDMetadata setUUIDMetadata();
+
+    @NotNull GetAllUUIDMetadata getAllUUIDMetadata();
+
+    @NotNull GetUUIDMetadata getUUIDMetadata();
+
+    @NotNull RemoveUUIDMetadata removeUUIDMetadata();
+
+    SetChannelMetadata.Builder setChannelMetadata();
+
+    @NotNull GetAllChannelsMetadata getAllChannelsMetadata();
+
+    @NotNull GetChannelMetadata.Builder getChannelMetadata();
+
+    RemoveChannelMetadata.Builder removeChannelMetadata();
+
+    @NotNull GetMemberships getMemberships();
+
+    @NotNull SetMemberships.Builder setMemberships();
+
+    RemoveMemberships.Builder removeMemberships();
+
+    ManageMemberships.Builder manageMemberships();
+
+    GetChannelMembers.Builder getChannelMembers();
+
+    SetChannelMembers.Builder setChannelMembers();
+
+    RemoveChannelMembers.Builder removeChannelMembers();
+
+    ManageChannelMembers.Builder manageChannelMembers();
+
+    @NotNull AddMessageAction addMessageAction();
+
+    @NotNull GetMessageActions getMessageActions();
+
+    @NotNull RemoveMessageAction removeMessageAction();
+
+    SendFile.Builder sendFile();
+
+    ListFiles.Builder listFiles();
+
+    GetFileUrl.Builder getFileUrl();
+
+    DownloadFile.Builder downloadFile();
+
+    DeleteFile.Builder deleteFile();
+
+    PublishFileMessage.Builder publishFileMessage();
+
+    void reconnect();
+
+    void disconnect();
+
+    @NotNull Publish fire();
+
+    @NotNull List<String> getSubscribedChannels();
+
+    @NotNull List<String> getSubscribedChannelGroups();
+
+    @NotNull Channel channel(@NotNull String name);
+
+    @NotNull ChannelGroup channelGroup(@NotNull String name);
+
+    @NotNull ChannelMetadata channelMetadata(@NotNull String id);
+
+    @NotNull UserMetadata userMetadata(@NotNull String id);
+
+    @NotNull SubscriptionSet subscriptionSetOf(@NotNull Set<? extends Subscription> subscriptions);
+
+    void addListener(@NotNull EventListener listener);
+
+    void addListener(@NotNull StatusListener listener);
+
+    void addListener(@NotNull SubscribeCallback listener);
+
+    InputStream decryptInputStream(InputStream inputStream) throws PubNubException;
+
+    @Nullable String encrypt(String inputString) throws PubNubException;
```

Comment:
On master we have also encrypt method with following signature:
`public String encrypt(String inputString, String cipherKey) throws PubNubException {`
What is a reason not to expose it here?

Reply:
yup